aslprep/workflows/asl/reference.py

```
@@ -130,14 +134,25 @@ def init_raw_aslref_wf(
         ]),
     ])  # fmt:skip
 
-    if m0scan:
+    if reference_volume_type == 'separate_m0scan':
         val_m0scan = pe.Node(
             ValidateImage(),
             name='val_m0scan',
             mem_gb=config.DEFAULT_MEMORY_MIN_GB,
         )
         workflow.connect([(inputnode, val_m0scan, [('m0scan', 'in_file')])])
 
+        # In some cases, the separate M0 scan may have a different resolution than the ASL scan.
+        # We resample the M0 scan to match the ASL scan at this point.
+        resample_m0scan_to_asl = pe.Node(
+            Resample(outputtype='NIFTI'),
+            name='resample_m0scan_to_asl',
+        )
+        workflow.connect([
+            (val_asl, resample_m0scan_to_asl, [('out_file', 'master')]),
+            (val_m0scan, resample_m0scan_to_asl, [('out_file', 'in_file')]),
+        ])  # fmt:skip
```

3dresample isn't in the Docker image. @mattcieslak should I just use antsApplyTransforms with an identity transform?

---

Maybe `nilearn.image.resample_to_img` would be enough.

---

antsApplyTransforms with 'identity' would work well too. I am now a big fan of Chris's recommendation for Gaussian resampling

---

Just switched over. Thanks!

---

It's currently failing because the M0 scan is 4D instead of 3D I guess. I'll figure out how to fix that tomorrow.

---

Or maybe it's because the reference image (the ASL file) was 4D.

---

You can set the input image type. This happens in qsiprep when we transform the cnr maps from eddy

---

The m0scans might be 3D or 4D. Won't explicitly setting the input image type cause a problem if the dimensionality doesn't match?

---

you can do this: https://github.com/PennLINC/qsiprep/blob/c74ff2f041a9ff6121042d30b5467af6082be85f/qsiprep/workflows/dwi/resampling.py#L195

---

That worked! Thanks!